packages/plugin/src/rules/naming-convention.ts

```diff
@@ -8,39 +8,78 @@ const formats = {
   UPPER_CASE: /^[A-Z_]*$/g,
 };
 
-function checkNameFormat(value, style, leadingUnderscore, trailingUnderscore) {
+type ValidNaming = 'camelCase' | 'PascalCase' | 'snake_case' | 'UPPER_CASE';
+const acceptedStyles: ValidNaming[] = ['camelCase', 'PascalCase', 'snake_case', 'UPPER_CASE'];
+interface CheckNameFormatParams {
+  value: string;
+  style?: ValidNaming;
+  leadingUnderscore: 'allow' | 'forbid';
+  trailingUnderscore: 'allow' | 'forbid';
+  suffix: string;
+}
+function checkNameFormat(params: CheckNameFormatParams): { ok: false; errorMessage: string } | { ok: true } {
+  const { value, style, leadingUnderscore, trailingUnderscore, suffix } = params;
   let name = value;
   if (leadingUnderscore === 'allow') {
     [, name] = name.match(/^_*(.*)$/);
   }
   if (trailingUnderscore === 'allow') {
     name = name.replace(/_*$/, '');
   }
-  return new RegExp(formats[style]).test(name);
+  if (suffix && !name.endsWith(suffix)) {
+    return { ok: false, errorMessage: '{{nodeType}} name "{{nodeName}}" should have "{{suffix}}" suffix' };
+  }
+  if (style && !acceptedStyles.some(acceptedStyle => acceptedStyle === style)) {
+    return {
+      ok: false,
+      errorMessage: `{{nodeType}} name "{{nodeName}}" should be in one of the following options: ${acceptedStyles.join(
+        ','
+      )}`,
+    };
+  }
+  if (!formats[style]) {
+    return { ok: true };
+  }
+  const ok = new RegExp(formats[style]).test(name);
+  if (ok) {
+    return { ok: true };
+  }
+  return { ok: false, errorMessage: '{{nodeType}} name "{{nodeName}}" should be in {{format}} format' };
 }
 
 const schemaOption = {
-  type: 'string',
-  enum: ['camelCase', 'PascalCase', 'snake_case', 'UPPER_CASE'],
+  type: ['string', 'object'],
```

Comment:
Since this is now an array, the enum option [here](https://github.com/dotansimha/graphql-eslint/pull/201/files#diff-9afe7600c3e6002f30eb08d02a06c30afda813baa781871f4b9c7af93039f2d6L24) no longer works and we have to validate  in `checkNameFormat `

Reply:
All good. I'm fine with breaking change at this point.